src/recordpage.cpp

```diff
@@ -664,6 +685,44 @@ void RecordPage::onOutputFileNameSelected(const QString& filePath)
     }
 }
 
+void RecordPage::onApplicationConfigAdded()
+{
+    saveApplicationConfig(QLatin1String("Config %1").arg(QString::number(ui->launchConfigComboBox->count() + 1)));
+
+    updateListOfApplicationConfigs();
+}
+
+void RecordPage::onApplicationConfigRenamed()
+{
+    auto group = applicationConfig(ui->applicationName->text());
+    const int index = ui->launchConfigComboBox->currentIndex();
+
+    // itemData is used to store the old name of the config
+    // after deleting the old config this must be changed to the new name
+    const auto oldText = ui->launchConfigComboBox->itemData(index).toString();
+
+    group.deleteGroup(oldText);
+
+    const auto currentText = ui->launchConfigComboBox->currentText();
+    saveApplicationConfig(currentText);
+
+    ui->launchConfigComboBox->setItemData(index, currentText);
+}
+
+void RecordPage::onApplicationConfigRemoved()
+{
+    auto appConfig = applicationConfig(ui->applicationName->text());
+    appConfig.deleteGroup(ui->launchConfigComboBox->currentText());
+
+    updateListOfApplicationConfigs();
+}
+
+void RecordPage::onApplicationConfigChanged(int index)
+{
+    Q_UNUSED(index);
```

Comment:
just remove the index? and I think such trivial one-liners could just be a lambda directly connected to the signal